iocore/net/quic/QUICFlowController.cc

```diff
@@ -217,39 +231,53 @@ QUICLocalFlowController::_need_to_forward_limit()
 QUICFrameUPtr
 QUICRemoteConnectionFlowController::_create_frame()
 {
-  auto frame                = QUICFrameFactory::create_blocked_frame(this->_offset, this->_issue_frame_id(), this);
-  QUICFrameInformation info = {frame->type(), QUICEncryptionLevel::NONE};
-  *(reinterpret_cast<QUICOffset *>(info.data)) = this->_offset;
-  this->_records_frame(frame->id(), info);
+  auto frame                    = QUICFrameFactory::create_blocked_frame(this->_offset, this->_issue_frame_id(), this);
+  QUICFrameInformationUPtr info = QUICFrameInformationUPtr(quicFrameInformationAllocator.alloc());
+  BlockedFrameInfo *frame_info  = reinterpret_cast<BlockedFrameInfo *>(info->data);
+  info->type                    = frame->type();
+  info->level                   = QUICEncryptionLevel::NONE;
+  frame_info->offset            = this->_offset;
+  this->_records_frame(frame->id(), std::move(info));
   return frame;
 }
 
 QUICFrameUPtr
 QUICLocalConnectionFlowController::_create_frame()
 {
-  auto frame                = QUICFrameFactory::create_max_data_frame(this->_limit, this->_issue_frame_id(), this);
-  QUICFrameInformation info = {frame->type(), QUICEncryptionLevel::NONE};
-  *(reinterpret_cast<QUICOffset *>(info.data)) = this->_limit;
-  this->_records_frame(frame->id(), info);
+  auto frame                    = QUICFrameFactory::create_max_data_frame(this->_limit, this->_issue_frame_id(), this);
+  QUICFrameInformationUPtr info = QUICFrameInformationUPtr(quicFrameInformationAllocator.alloc());
+  MaxDataFrameInfo *frame_info  = reinterpret_cast<MaxDataFrameInfo *>(info->data);
+  info->type                    = frame->type();
+  info->level                   = QUICEncryptionLevel::NONE;
+  frame_info->maximum_data      = this->_limit;
+  this->_records_frame(frame->id(), std::move(info));
   return frame;
 }
 
 QUICFrameUPtr
 QUICRemoteStreamFlowController::_create_frame()
 {
   auto frame = QUICFrameFactory::create_stream_blocked_frame(this->_stream_id, this->_offset, this->_issue_frame_id(), this);
-  QUICFrameInformation info                    = {frame->type(), QUICEncryptionLevel::NONE};
-  *(reinterpret_cast<QUICOffset *>(info.data)) = this->_offset;
-  this->_records_frame(frame->id(), info);
+  QUICFrameInformationUPtr info      = QUICFrameInformationUPtr(quicFrameInformationAllocator.alloc());
+  StreamBlockedFrameInfo *frame_info = reinterpret_cast<StreamBlockedFrameInfo *>(info->data);
+  info->type                         = frame->type();
+  info->level                        = QUICEncryptionLevel::NONE;
+  frame_info->stream_id              = this->_stream_id;
```

Comment:
We don't need to record stream id, because FlowController class regenerate the frame and it knows the stream id. The original code should satisfy everything (we don't need StreamBlockedFrameInfo and BlockedFrameInfo).

Reply:
Let's go back to the original code .